lib/sisimai/bite/email/biglobe.rb

```diff
@@ -0,0 +1,148 @@
+module Sisimai::Bite::Email
+  module Biglobe
+    # Sisimai::Bite::Email::Biglobe parses a bounce email which created by BIGLOBE.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/Biglobe.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :subject => %r/\AReturned mail:/,
+        :from    => %r/postmaster[@](?:biglobe|inacatv|tmtv|ttv)[.]ne[.]jp/,
+      }
+      Re1 = {
+        :begin  => %r/\A   ----- The following addresses had delivery problems -----\z/,
+        :error  => %r/\A   ----- Non-delivered information -----\z/,
+        :rfc822 => %r|\AContent-Type: message/rfc822\z|,
+        :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ReFailure = {
+        filtered:    %r/Mail Delivery Failed[.]+ User unknown/,
+        mailboxfull: %r/The number of messages in recipient's mailbox exceeded the local limit[.]/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'BIGLOBE: http://www.biglobe.ne.jp'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return []; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Biglobe
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [64.6/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->